test/langtools/tools/javac/diags/examples/NotAnInterfaceComponent.java

```diff
@@ -21,9 +21,9 @@
  * questions.
  */
 
-// key: compiler.misc.not.a.functional.intf.1
+// key: compiler.misc.bad.intersection.target.for.functional.expr
+// key: compiler.misc.not.an.intf.component
 // key: compiler.err.prob.found.req
-// key: compiler.misc.incompatible.abstracts
 
 class NotAnInterfaceComponent {
     Object o = (String & Runnable) ()-> { };
```

Comment:
How was this rejected before? The generated compiler keys look strange...

Reply:
it was rejected because javac was trying to determine the function descriptor associated with the supposed functional interface that could be derived from the intersection of `String & Runnable` a this is impossible then it was failing. This check is at `DescriptorCache::findDescriptorInternal`. But if we have an empty class as in the test case that shows the bug that only inherits from Object but doesn't defines any other method, then javac was only seeing the method defined in the interface and thus a function descriptor was "found"

Reply:
I see - I did some debugging, and the issue with `String` is that there's more abstracts methods like `describeConstable` -- which then trigger a compilation error. I agree that it's better to validate the intersection type upfront.